cirq/ops/moment.py

```diff
@@ -87,9 +87,17 @@ def __init__(self, *contents: 'cirq.OP_TREE') -> None:
 
         self._operations = tuple(op_tree.flatten_to_ops(contents))
 
-        # Check that operations don't overlap.
-        affected_qubits = [q for op in self.operations for q in op.qubits]
+        affected_qubits = []
```

Comment:
You can check that operations don't overlap by using the keys of `self._qubit_to_op` instead of constructing an additional list, `affected_qubits`

Reply:
Done, much cleaner now!